presto-main/src/main/java/com/facebook/presto/sql/planner/ExpressionInterpreter.java

```diff
@@ -478,18 +479,17 @@ protected Object visitInPredicate(InPredicate node, Object context)
             }
             InListExpression valueList = (InListExpression) valueListExpression;
 
-            Set<Object> set = inListCache.get(valueList);
+            Set<?> set = inListCache.get(valueList);
 
             // We use the presence of the node in the map to indicate that we've already done
             // the analysis below. If the value is null, it means that we can't apply the HashSet
             // optimization
             if (!inListCache.containsKey(valueList)) {
                 if (Iterables.all(valueList.getValues(), ExpressionInterpreter::isNullLiteral)) {
+                    HashSet<Object> objectSet = new HashSet<>();
                     // if all elements are constant, create a set with them
-                    set = new HashSet<>();
-                    for (Expression expression : valueList.getValues()) {
-                        set.add(process(expression, context));
-                    }
+                    objectSet.addAll(valueList.getValues().stream().map(expression -> process(expression, context)).collect(Collectors.toList()));
```

Comment:
instead of collecting to a List and then adding to the set, just directly collect it to a Set